internal/cli/auditing/update.go

```diff
@@ -0,0 +1,126 @@
+	auditFilter               string
+	fs                        afero.Fs
+	store                     store.AuditingUpdater
+}
+
+func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var updateTemplate = "Auditing configuration successfully updated.\n"
+
+func (opts *UpdateOpts) Run() error {
+	auditLog, err := opts.newAuditLog()
+	if err != nil {
+		return err
+	}
+	r, err := opts.store.UpdateAuditingConfig(opts.ConfigProjectID(), auditLog)
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newAuditLog() (*atlasv2.AuditLog, error) {
+	if opts.filename != "" {
+		fileContent, err := file.LoadFile(opts.fs, opts.filename)
+		if err != nil {
+			return nil, err
+		}
+		opts.auditFilter = string(fileContent)
+	}
+
+	return &atlasv2.AuditLog{
+		Enabled:                   &opts.enabled,
+		AuditAuthorizationSuccess: &opts.auditAuthorizationSuccess,
+		AuditFilter:               &opts.auditFilter,
+	}, nil
+}
+
+// UpdateBuilder atlas auditing update [--file filter.json] [--auditFilter "{"atype": "authenticate"}"] [--auditAuthorizationSuccess] [--enabled] --projectId projectId.
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{
+		fs: afero.NewOsFs(),
+	}
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Updates the auditing configuration for the specified project",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Project Owner"),
+		Args:  require.NoArgs,
+		Annotations: map[string]string{
+			"output": updateTemplate,
+		},
+		Example: `  # Audit all authentication events for known users:
+  atlas auditing update --auditFilter "{"atype": "authenticate"}"
+
+  # Audit all authentication events for known user via a configuration file:
+  atlas auditing update -f filter.json
+`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return opts.PreRunE(
+				opts.ValidateProjectID,
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), updateTemplate),
+			)
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVar(&opts.auditFilter, flag.AuditFilter, "", usage.AuditFilter)
+	cmd.Flags().BoolVar(&opts.enabled, flag.Enabled, false, usage.EnabledAuditing)
+	cmd.Flags().StringVarP(&opts.filename, flag.File, flag.FileShort, "", usage.AuditingFilename)
+	cmd.Flags().BoolVar(&opts.auditAuthorizationSuccess, flag.AuditAuthorizationSuccess, false, usage.AuditAuthorizationSuccess)
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+	_ = cmd.RegisterFlagCompletionFunc(flag.Output, opts.AutoCompleteOutputFlag())
+
+	_ = cmd.MarkFlagFilename(flag.File)
+
+	cmd.MarkFlagsMutuallyExclusive(flag.File, flag.AuditFilter)
```

Comment:
the flags are mutually exclusive but I don't see an `opts.validateFlags` to check if one of the 2 is set

Reply:
good point 👍

Reply:
updated